### DISCUSSION TITLE
find file ignoring filters

### DISCUSSION THREAD

**Discussion by przepompownia** [nvim-tree/nvim-tree.lua#1703] [Ideas]
Currently I have a local function to open nvim-tree with the current file and root resolved at the top level of it's git repo (or CWD otherwise).
```lua
local treeapi = require('nvim-tree.api')
-- ...
local function focusOnFile()
  local bufName = vim.api.nvim_buf_get_name(0)
  treeapi.tree.open(git.top(vim.fs.dirname(bufName)))
  --  treeapi.tree.toggle_hidden_filter()
  treeapi.tree.find_file(bufName)
end

keymap.set({'n'}, '<Plug>(ide-tree-focus-current-file)', focusOnFile)
```
(full source: https://github.com/przepompownia/vim-arctgx/blob/master/bundleConfig/nvim-tree.lua).

If the current file is under hidden directory and filter for `dotfiles` is set to `true` by config or `H`, then `find_file(bufName)` does not reveal it's node.

Is it possible to force reveal the node hidden by filters?

**Comment by alex-courtis** [comment]
Displaying a file that is hidden by any of `nvim-tree.filters` is not currently possible without toggling the filters.

I can see many possibilities for behaviour that would allow this, some complex to implement, some simple. Example: "finding a file that is currently hidden by a filter toggles all the filters off".

Please describe some behaviours that would match your use case and expectations.

**Reply by przepompownia** [reply]
Maybe one-time switching off all the filters that hide the selected file (please note that I don't need to search by some part of a path) would be enough (e.g. if the path would be under dotfiles and not git-ignored, only one filter would be need to disable). 

The common case is if I start nvim-tree from some dotfile under my home directory using the above mapping. While bypassing filters is not possible, I switched `dotfiles` to `false`.

**Reply by alex-courtis** [reply]
Thank you. Let's continue in #1587

**Comment by alex-courtis** [comment]
See #1587 for similar discussions.
